src/profile-logic/import/instruments/BinaryPlistParser.js

```diff
@@ -0,0 +1,196 @@
+
+    return new Uint8Array(this.view.buffer, offset + both.offset, both.length);
+  }
+
+  parseStringASCII(offset: number, extra: number): string {
+    const both = this.parseLengthAndOffset(offset, extra);
+    let text = '';
+    offset += both.offset;
+    for (let i = 0; i < both.length; i++) {
+      text += String.fromCharCode(this.view.getUint8(offset++));
+    }
+    return text;
+  }
+
+  parseStringUTF16(offset: number, extra: number): string {
+    const both = this.parseLengthAndOffset(offset, extra);
+    let text = '';
+    offset += both.offset;
+    for (let i = 0; i < both.length; i++) {
+      text += String.fromCharCode(this.view.getUint16(offset, false));
+      offset += 2;
+    }
+    return text;
+  }
+
+  parseUID(offset: number, size: number): UID {
+    return new UID(this.parseInteger(offset, size));
+  }
+
+  parseArray(offset: number, extra: number): any[] {
+    const both = this.parseLengthAndOffset(offset, extra);
+    const array: any[] = [];
+    const size = this.referenceSize;
+    offset += both.offset;
+    for (let i = 0; i < both.length; i++) {
+      array.push(
+        this.parseObject(this.offsetTable[this.parseInteger(offset, size)])
+      );
+      offset += size;
+    }
+    return array;
+  }
+
+  parseDictionary(offset: number, extra: number): Object {
+    const both = this.parseLengthAndOffset(offset, extra);
+    const dictionary = Object.create(null);
+    const size = this.referenceSize;
+    let nextKey = offset + both.offset;
+    let nextValue = nextKey + both.length * size;
+    for (let i = 0; i < both.length; i++) {
+      const key = this.parseObject(
+        this.offsetTable[this.parseInteger(nextKey, size)]
+      );
+      const value = this.parseObject(
+        this.offsetTable[this.parseInteger(nextValue, size)]
+      );
+      if (typeof key !== 'string')
+        throw new Error('Unexpected non-string key at offset ' + nextKey);
+      dictionary[key] = value;
+      nextKey += size;
+      nextValue += size;
+    }
+    return dictionary;
+  }
+
+  parseObject(offset: number): any {
+    const marker = this.view.getUint8(offset++);
+    const extra = marker & 0x0f;
+    switch (marker >> 4) {
+      case 0x0:
+        return this.parseSingleton(offset, extra);
+      case 0x1:
+        return this.parseInteger(offset, 1 << extra);
+      case 0x2:
+        return this.parseFloat(offset, 1 << extra);
+      case 0x3:
+        return this.parseDate(offset, 1 << extra);
+      case 0x4:
+        return this.parseData(offset, extra);
+      case 0x5:
+        return this.parseStringASCII(offset, extra);
+      case 0x6:
+        return this.parseStringUTF16(offset, extra);
+      case 0x8:
+        return this.parseUID(offset, extra + 1);
+      case 0xa:
+        return this.parseArray(offset, extra);
+      case 0xd:
+        return this.parseDictionary(offset, extra);
+    }
+    throw new Error('Unexpected marker ' + marker + ' at offset ' + --offset);
```

Comment:
This switch statement should have a default case, we can throw this error there.